### ISSUE TITLE
Please remove all Google fonts references from source code and replace it with local fonts

### ISSUE THREAD

**Issue by saschafoerster** [pressbooks/ideas#408]
Google fonts can't be used in the EU anymore, without risks: 
https://www.golem.de/news/landgericht-muenchen-einbindung-von-google-fonts-ist-rechtswidrig-2202-162826.html

So please make it easy for us and remove Google Fonts reverences and use instead fonts locally.

**Comment by SteelWagstaff** [comment]
Thanks @saschafoerster -- we will look into this soon. For those who don't read German, here's a summary of the article he shared:
A Munich Regional Court judge has ruled that the integration of dynamic web content such as Google Fonts from US web services is illegal without the consent of the visitor. The ruling stated that there is no legitimate interest within the meaning of the General Data Protection Regulation (GDPR), because the Google Fonts can be downloaded and delivered from your own server instead of integrating them via external Google servers. The website operator faces a fine of up to 250,000 euros or imprisonment for up to six months if the plaintiff's IP address is passed on to Google when the website is visited.

While this judgement is about the integration of Google fonts specifically, the principles established by the court also apply to all other content from US services integrated externally into websites. If the ruling stands, any content served by a CDN would be affected. To avoid violating the law, websites must host content such as fonts, scripts or images themselves. Alternatively, website hosts could obtain consent to the disclosure of the IP address via a consent banner.

Here's a link to the judgment itself: https://rewis.io/urteile/urteil/lhm-20-01-2022-3-o-1749320/ (in German)

**Comment by saschafoerster** [comment]
@SteelWagstaff Thank you very much for helping with translation and the very fast reaction!

**Comment by SteelWagstaff** [comment]
See https://discourse.roots.io/t/guide-self-hosted-google-fonts/22794

**Comment by SteelWagstaff** [comment]
Still relevant. FYI @cmurtagh

**Comment by cmurtagh** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pressbooks-Product-Backlog-60ae7d807f6e370010b3d5be/issues/pressbooks/ideas/408?planning-poker) with Zenhub @arzola @fdalcin @ho-man-chan @MichelleWeremczuk @richard015ar @tw77

**Comment by fdalcin** [comment]
research whether we can use https://fonts.bunny.net/ as an option

**Comment by saschafoerster** [comment]
To be clear what I meant:  I did not propose to change one provider from the US versus another provider from the US. All of them have the same issue, if they collect data or not, the IP is transferred with every single request. Their policies can change, I am not under control. 
My wish is to have NO third-party at all, unless I explicitly allow and want this. I want to upload fonts in the backend or via CSS myself. It should only be an option to download fonts from a third party and then cache them locally on my server. And only I explicitly agree, then a third party domain can be used for caching (like Google or Bunny.net). 
Same is of course true for all other libraries or javascripts. :)

**Comment by richard015ar** [comment]
I have invested some effort in this task. After an initial research and a conversation with devs, I figured we have a few options to approach this:
- Download all the possible fonts to use in our repos (see an outdated and incomplete list at: https://github.com/pressbooks/pressbooks-book/blob/dev/readme-fonts.txt) and add them to our repos. After that, re-write css pointing to remote URLs and point them to a path instead. This is the least efficient and maintainable approach.
- Download fonts on demand, as some popular plugins do, such as [Local Google Fonts](https://wordpress.org/plugins/local-google-fonts/). It is only for Google Fonts, in our case some of the available fonts in PB are from other different sources as well. It uses some background processes to detect google fonts and uses some hooks to download and replace them. This plugin does not work as expected for multisite installation as PB, unfortunately. However a similar approach for a separate plugin could be considered.
- Use a public repo to host all our supported fonts, host them in a separate service and provide them on demand, for each network. 
- Use just a few fonts in our repo, and allow our users through an admin interface in a separate plugin to upload their own fonts.

This lib: https://discourse.roots.io/t/guide-self-hosted-google-fonts/22794 as others such as google fonts helpers: https://github.com/majodev/google-webfonts-helper requires us to download the fonts (first items of the list of approaches). But those are handy and could be considered if we decide taking that path.

The final solution of this, will likely to require a new development and there is no an easy solution as we thought I'm afraid. In any case a consultation with Ned is commendable. 

Next steps: we'll have a conversation with Christopher to plan a meeting with Ned. After that, we'll decide in a path, refine and explore it.

**Comment by cmurtagh** [comment]
We will schedule a conversation with devs and Ned to come up with a strategy so that we can size this appropriately.

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 4y 1m ago with no updates in the last 1y 1m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542

**Comment by saschafoerster** [comment]
GDPR is still relevant. And the more in the actual world.

**Comment by arzola** [comment]
Agreed, I'll re open this one and place it into our current backlog

**Comment by MichSpieler** [comment]
It would be great to include this request at the same time: https://github.com/pressbooks/pressbooks/issues/4090#issuecomment-3559989911

**Comment by SteelWagstaff** [comment]
We may want to look into this plugin: https://wordpress.org/plugins/host-webfonts-local/

**Comment by SteelWagstaff** [comment]
@MichSpieler @arzola I created this issue for the Noto Fonts request: https://github.com/pressbooks/pressbooks-book/issues/1413
